Refactor few code style rules to separate docs (#20856)
dotnet/docs: docs/fundamentals/code-analysis/style-rules/ide0044.md

---
title: "IDE0044: Add readonly modifier"
description: "Learn about code analysis rule IDE0044: Add readonly modifier"
ms.date: 09/30/2020
ms.topic: reference
f1_keywords:
- IDE0044
- dotnet_style_readonly_field
helpviewer_keywords:
- IDE0044
- dotnet_style_readonly_field
author: gewarren
ms.author: gewarren
dev_langs:
- CSharp
- VB
---
# Add readonly modifier (IDE0044)

|Property|Value|
|-|-|
| **Rule ID** | IDE0044 |
| **Title** | Add readonly modifier |
| **Category** | Style |
| **Subcategory** | Language rules (modifier preferences) |
| **Applicable languages** | C# and Visual Basic |
| **Introduced version** | Visual Studio 2017 version 15.7 |

## Overview

> This style rule concerns specifying the readonly modifier for fields which are only initialized (either inline or inside of a constructor), but never re-assigned.

Reviewer: ```suggestion
This style rule concerns specifying the readonly modifier for fields that are initialized (either inline or inside of a constructor) but never reassigned.
```
